### ISSUE TITLE
It should also treat the filename and folder name length 

### ISSUE THREAD

**Issue by conualfy** [murukeshm/scripts#1]
Because EXT filesystem allows longer names than NTFS, this should also be taken into consideration.   If there will be more files in the same folder to be trunked, they need to be somehow differentiated so they will not be overwritten; a solution is to add numbers in the end in that case, allowing, let's say, for at least 100 in the same folder -00... -99

Optional, to cover all possible cases: if more (not that many chances, but could happen), another letter should be removed and should use 3 digits&so on.

**Comment by muru** [comment]
I looked up the limits on [Wikipedia](https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits). It seems NTFS and ext4 both allow the same maximum component length: 255. The difference seems to be in the maximum path length (32k for NTFS, not limited in ext4), but here [Windows adds an additional wrinkle](https://stackoverflow.com/a/265782/2072269) with a `MAX_PATH` of  259 chars + NUL.

Which limit do we go with? 32k is effectively large enough that I don't think anyone using this would run into it. 259 is small enough I'd think I might run into it; but then the effective length from the mount point needs to be considered.

**Comment by conualfy** [comment]
Yes, it happens. I did encounter it a few times copying large folders with
many intricate folders with long names.
Last time: today. I want to back-up my faculty folder and to keep track of
things, I rename folders and files to match the content ("-" instead of
space).

F.
--
VP Romanian National Guides' Federation
drumliber.ro, vezi Romania
Bucharest.Guide, ghid în București
Primul blogger de travel premiat la Gala AJTR - 15 ianuarie 2014
obiectiveturistice.drumliber.ro - proiect premiat la Gala Webstock 2012
Duminica la ora 16:10 la Radio România Actualități / Ca Pe Roate

mar., 31 mar. 2020, 05:41 Murukesh Mohanan <notifications@github.com> a
scris:
